wordle-app/scripts/game.ts

```diff
@@ -0,0 +1,116 @@
+import { Axios } from "axios";
+import { LetterState, type Letter } from "./letter";
+import { Word } from "./word";
+import { WordList } from "./wordList";
+import nameUserNameDialog from "../pages/index.vue"
+import stopwatch from "../pages/index.vue"
+
+export class Game {
+  public maxAttempts: number;
+  public guesses: Word[] = [];
+  public secretWord: string = "";
+  public guessIndex: number = 0;
+  public gameState: GameState = GameState.Playing;
+  public guessedLetters: Letter[] = [];
+
+  constructor( secretWord: string, maxAttempts: number = 6) {
+    this.maxAttempts = maxAttempts;
+    this.secretWord = secretWord.toUpperCase();
+    this.startNewGame();
+  }
+
+  public startNewGame() {
+    this.guessIndex = 0;
+    this.gameState = GameState.Playing;
+    this.guessedLetters = [];
+
+    // Populate guesses with the correct number of empty words
+    this.guesses = [];
+    for (let i = 0; i < this.maxAttempts; i++) {
+      this.guesses.push(
+        new Word({ maxNumberOfLetters: this.secretWord.length })
+      );
+    }
+  }
+
+  public get guess() {
+    return this.guesses[this.guessIndex];
+  }
+
+  public removeLastLetter() {
+    if (this.gameState === GameState.Playing) {
+      this.guess.removeLastLetter();
+    }
+  }
+
+  public addLetter(letter: string) {
+    if (this.gameState === GameState.Playing) {
+      this.guess.addLetter(letter);
+    }
+  }
+
+  public updateGuessedLetters() {
+    for (const letter of this.guess.letters) {
+      // Find the index of the letter in the guessed letters array
+      const index = this.guessedLetters.findIndex(
+        (existingLetter) => existingLetter.char === letter.char
+      );
+      if (index !== -1) {
+        // Do not update the letter if it is already correct
+        if (this.guessedLetters[index].state !== LetterState.Correct) {
+          // Do not update the letter if it is wrong
+          if (letter.state !== LetterState.Wrong) {
+            this.guessedLetters[index] = letter;
+          }
+        }
+      } else {
+        // If letter does not already exist, add it to the array
+        this.guessedLetters.push(letter);
+      }
+    }
+  }
+
+  public submitGuess() {
+    if (this.gameState !== GameState.Playing) return;
+    if (!this.guess.isFilled) return;
+    if (!this.guess.isValidWord()) {
+      this.guess.clear();
+      return;
+    }
+
+    const isCorrect = this.guess.compare(this.secretWord);
+    this.updateGuessedLetters();
+
+    if (isCorrect) {
+      this.gameState = GameState.Won;
+      this.postScore();
+    } else {
+      if (this.guessIndex === this.maxAttempts - 1) {
+        this.gameState = GameState.Lost;
+      } else {
+        this.guessIndex++;
+      }
+    }
+  }
+  public postScore() {
+    let attempts = 0;
+    if (this.gameState == GameState.Won) {
+      attempts = this.guessIndex + 1;
+    } else {
+      attempts = this.maxAttempts;
+    }
+    //TODO - INSERT API LINK
```

Comment:
Just leaving a comment that I saw your progress on the post score requirement. Sorry about all the red X's in the main review comment. You got this!

Reply:
No worries, this is not a finished assignment so it is on us.